notebooks/402-pose-estimation-webcam/402-pose-estimation.ipynb

```diff
@@ -0,0 +1,488 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# initialize inference engine\n",
+    "ie_core = ie.IECore()\n",
+    "# read the network and corresponding weights from file\n",
+    "net = ie_core.read_network(model=model_path, weights=model_weights_path)\n",
+    "# load the model on the CPU (you can use GPU or MYRIAD as well)\n",
+    "exec_net = ie_core.load_network(net, \"CPU\")\n",
+    "\n",
+    "# get input and output names of nodes\n",
+    "input_key = list(exec_net.input_info)[0]\n",
+    "output_keys = list(exec_net.outputs.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Input key is the name of the input node and output keys are names of output nodes of the network. In the case of the OpenPose Model, we have one input and two outputs: pafs and keypoints heatmap."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "input_key, output_keys"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### OpenPoseDecoder\n",
+    "\n",
+    "We need a decoder to transform raw results from the neural network into poses. This magic happens inside Open Pose Decoder. It comes from [OpenVINO Model Zoo](https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/common/python/models/open_pose.py) and it's suitable for the \"human-pose-estimation-0001\".\n",
+    "\n",
+    "If you choose a model other than \"human-pose-estimation-0001\" you will need another decoder (e.g. AssociativeEmbeddingDecoder), which is available in [OpenVINO Model Zoo](https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/common/python/models/hpe_associative_embedding.py) as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "decoder = OpenPoseDecoder()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Result processing\n",
+    "\n",
+    "A bunch of useful functions to transform results into poses.\n",
+    "\n",
+    "Firstly we pool heatmap. As pooling is not available in numpy, we use a simple trick to do that directly in numpy. Then we use non-maximum suppression to get keypoints from the heatmap. After that, we decode poses using the created decoder. As the input image is bigger than network outputs, we need to multiply all pose coordinates by the scaling factor."
```

Comment:
```suggestion
    "First, we will pool the heatmap. Since pooling is not available in numpy, we use a simple method to do it directly with numpy. Then, we use non-maximum suppression to get the keypoints from the heatmap. After that, we decode poses using the decoder. Since the input image is bigger than the network outputs, we need to multiply all pose coordinates by a scaling factor."
```